QuadradiusR/scripts/torus.gd

```diff
@@ -140,8 +191,67 @@ func make_move(source_tile: Tile, target_tile: Tile, is_colliding: bool = false)
 # OR destroy_piece(torus) which would make one piece destroy the other, so it could handle
 # all the consequences
 
+# # # # # # #
+# Powerups  #
+# # # # # # #
+func add_power(power_id: String, power_def_id):
+	if powerup_id_to_definition_id.empty():
+		anim.queue("Tint" + player_id)
+	powerup_id_to_definition_id[power_id] = power_def_id
+
+	if not power_def_id: # the client is not permitted to view the power
+		return
+
+	if not powerup_definition_id_to_count.has(power_def_id):
+		powerup_definition_id_to_count[power_def_id] = 1
+		powers_list.add_powerup(power_def_id, Context.get_power_by_definition_id(power_def_id)['name'])
+	else:
+		powerup_definition_id_to_count[power_def_id] += 1
+		powers_list.add_powerup_charges(power_def_id, 1)
+
+
+func use_power(power_id: String):
+	powers_list.use_power(powerup_id_to_definition_id[power_id])
+	# TODO Actually use the power, send data to server etc.
```

Comment:
this should probably be implemented (unless it was moved to another issue but idk), should use the api method to send the info about powerup used. Its effects can either be processed immediately (right now only raise tile is added) or it can be done after receiving gamestate (will be cleaner and will work both for own changes and opponent's)